### ISSUE TITLE
Support multiprocessing with Web Workers

### ISSUE THREAD

**Issue by parasyte** [melonjs/melonJS#459]
If we can parallelize parts of the update loop, then we could vastly increase performance by doing most of the work in worker threads.

A simple example would be doing all AI updates in one thread, and simultaneously doing collision detection in another thread.

A very complex example would be doing 1/8 of the collision detection in each thread, with a total of 8 threads (optimal configuration for an 8-core machine). The biggest issue here is doing parallel collision detection/response in a deterministic manner.

**Comment by parasyte** [comment]
@Zolmeister zed.js looks pretty reasonable!

It reminds me of a stack implementation that I started for WebGL (but did not use). It was a wrapper around a typed array that would grow as necessary (only exponential growth) and never shrink. Which is good for a stack because you're constantly pushing and popping from the array. It didn't make much difference with performance, because V8 is pretty smart about small arrays. Would have been better suited for stacks that grow and shrink rapidly.

I noticed your string encoding uses Uint8, but JS Strings internally use UTF-16; `charCodeAt` returns a 16-bit number. It should also be possible to replace your read-loop https://github.com/Zolmeister/zed.js/blob/master/index.js#L116-L118 with something like:

```javascript
return String.fromCharCode.apply(String, Array.apply([], this.uint16View.subarray(index / 2 + 1)));
```

And similar for the write-loop:

```javascript
this.uint16View.set(val.split("").map(function (v) { return v.charCodeAt(0); }), index / 2  + 1);
```

That should make the string encoding a bit more efficient and correct. :)

Also, be careful about special number values, like `NaN` and `Infinity`.
